packages/cli/lib/lib/webpack/prerender.js

```diff
@@ -24,10 +24,15 @@ module.exports = function(env, params) {
 			);
 			return '';
 		}
-
-		let preact = require('preact'),
-			renderToString = require('preact-render-to-string');
-
+		const { cwd } = env;
+		const preactPath = require.resolve(`${cwd}/node_modules/preact`);
+		const renderToStringPath = require.resolve(
+			`${cwd}/node_modules/preact-render-to-string`
+		);
+		let preact = require(preactPath),
+			renderToString = require(existsSync(renderToStringPath)
+				? renderToStringPath
+				: 'preact-render-to-string');
```

Comment:
kinda wondering if we actually need this, it should already resolve from cwd?

and we don't have it as a fixed dep anymore https://github.com/developit/preact-cli/blob/master/packages/cli/package.json#L64-L67

Reply:
preact/preact-render-to-string both should be used from same source otherwise `__hooks` fail.

Thats why we need to force require this from user land. This will allow user to have the flexibility of using any version of preact/render-to-string they want.

If they don't have, then we can fallback to `cwd` but any usage of hooks will fail.

This'll make it less of a breaking change for everyone as the old ones continue to work as is and the new one will install one more dep and can use hooks

Reply:
yeah I'm saying it should only exist in user land either way

Reply:
that breaks old users, no? just trying to minimize the breaking change impact

Reply:
it's already this way for cli v3

Reply:
well... its also not used anymore.. so ummm makes sense, will make the change

Reply:
I meant we shouldn't need the `${cwd}` part? or am I wrong?

Reply:
we actually do need ${cwd}. Otherwise it uses `preact/render-to-string` from preact-cli's `node_modules` folder instead of the created app's folder

Reply:
Also `${cwd}` points to the working directory of the created project and not the where `preact-cli` is installed globally.

Reply:
well `preact-cli` doesn't have them anymore

but yes it could get in the way if you're not using it via an npm script and or it's just installed globally

Reply:
Oh sorry for not conveying that.. in my head everyone uses it like that.

Anyways, i'll fix the test

Reply:
yeah I kinda assumed having it locally installed, you'll want that for automated builds either way